### DISCUSSION TITLE
Is `send(_:while:)` still relevant?

### DISCUSSION THREAD

**Discussion by davdroman** [pointfreeco/swift-composable-architecture#1719] [Q&A] [answered]
I noticed the `send(_:while:)` method on `ViewStore` is not deprecated, however if memory serves it was introduced before the recent Concurrency changes.

Is there a reason to keep this method around now that we can call `send` with `.finish()`?

**Comment by stephencelis** [comment, marked as answer]
@davdroman We think having an alternate convenience method for driving async off state can still be useful, though we're open to discussing further if folks feel like slimming the public API down would be helpful, especially for 1.0.

**Reply by davdroman** [reply]
Personally I think it could be misconstrued as an `Effect.run`-compatible API, which is not.

In fact just the other day I tried using it with an action that returned a `.run` effect, resulting in a crash, which is actually what has led me to open this discussion.

**Reply by jshier** [reply]
There shouldn't be any issue with `Effect.run`. We use it with that just fine, to drive loading indicators during network requests. `send(_:while:)` is actually one of the great TCA conveniences that nicely demonstrates the "magic" the framework enables.

**Reply by davdroman** [reply]
Is that so? My mistake then! It's quite strange, I got an EXC_BAD_ACCESS. It must've been something else I guess.

**Reply by stephencelis** [reply]
@davdroman Think you can repro? That definitely sounds strange so if we have a bug we should def look into it!
